internal/backend/gdrive/gdrive.go

```diff
@@ -0,0 +1,566 @@
+// repository.
+func (be *gdriveBackend) Location() string {
+	return be.Layout.(*backend.DefaultLayout).Path
+}
+
+// Test a boolean value whether a File with the name and type exists.
+func (be *gdriveBackend) Test(ctx context.Context, h restic.Handle) (bool, error) {
+	be.sem.GetToken()
+	defer be.sem.ReleaseToken()
+
+	_, err := be.getFile(ctx, h)
+	if err != nil && be.IsNotExist(err) {
+		return false, nil
+	}
+	return err == nil, err
+}
+
+// Remove removes a File described  by h.
+func (be *gdriveBackend) Remove(ctx context.Context, f restic.Handle) error {
+	be.sem.GetToken()
+	defer be.sem.ReleaseToken()
+
+	return be.removeItem(ctx, be.Filename(f))
+}
+
+// Close the backend, does nothing
+func (be *gdriveBackend) Close() error {
+	return nil
+}
+
+// Load returns a reader that yields the contents of the file at h at the
+// given offset. If length is larger than zero, only a portion of the file
+// is returned. rd must be closed after use. If an error is returned, the
+// ReadCloser must be nil.
+func (be *gdriveBackend) Load(ctx context.Context, h restic.Handle, length int, offset int64) (io.ReadCloser, error) {
+	be.sem.GetToken()
+
+	file, err := be.getFile(ctx, h)
+	if err != nil {
+		be.sem.ReleaseToken()
+		return nil, err
+	}
+
+	rd, err := gdriveGetFileReader(ctx, be.service, file.Id, length, offset)
+	if err != nil {
+		be.sem.ReleaseToken()
+		return nil, err
+	}
+
+	return be.sem.ReleaseTokenOnClose(rd, nil), nil
+}
+
+// Stat returns information about the File identified by h.
+func (be *gdriveBackend) Stat(ctx context.Context, h restic.Handle) (restic.FileInfo, error) {
+	be.sem.GetToken()
+	defer be.sem.ReleaseToken()
+
+	file, err := be.getFile(ctx, h)
+	if err != nil {
+		return restic.FileInfo{}, err
+	}
+	return restic.FileInfo{Size: file.Size}, nil
+}
+
+// List returns a channel that yields all names of files of type t in an
+// arbitrary order. A goroutine is started for this, which is stopped when
+// ctx is cancelled.
+func (be *gdriveBackend) List(ctx context.Context, t restic.FileType) <-chan string {
+	ch := make(chan string)
+
+	resultForwarder := func(item *drive.File) bool {
+		select {
+		case ch <- item.Name:
+			return true
+		case <-ctx.Done():
+			return false
+		}
+	}
+
+	listChildren := func(path string, consumer func(item *drive.File) bool) error {
+		be.sem.GetToken()
+		defer be.sem.ReleaseToken()
+
+		id, err := be.getFolderID(ctx, path)
+		if err != nil {
+			return err
+		}
+
+		return gdriveListFolder(ctx, be.service, id, consumer)
+	}
+
+	go func() {
+		defer close(ch)
+
+		prefix, hasSubdirs := be.Basedir(t)
+
+		var err error
+		if !hasSubdirs {
+			err = listChildren(prefix, resultForwarder)
+		} else {
+			subdirs := map[string]bool{}
+			err = listChildren(prefix, func(item *drive.File) bool { subdirs[item.Name] = true; return true })
+			if err == nil {
+				for subdir := range subdirs {
+					err = listChildren(prefix+"/"+subdir, resultForwarder)
```

Comment:
Don't concatenate strings, use `path.Join()` for that (OS-independent, always uses slashes).